### PR TITLE
Fix search by upgrading dependencies

### PR DESCRIPTION
**Pull request by pavelloz**
This PR fixes search by:
* upgrading ruby runtime to 3.2.2
* upgrading `just-docs-theme` to the newest (0.6.2) version, following the official guide (https://just-the-docs.com/changelog/#using-release-v060) and pinning it

<details>

<summary>Video after the fix</summary>

https://github.com/parallel-markets/handbook/assets/546845/a240129e-6294-488f-969d-592a13ce9b5d

</details>

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -1,7 +1,7 @@
 source "https://rubygems.org"
 gem 'github-pages', group: :jekyll_plugins
 gem "jekyll-minifier", "~> 0.1"
-gem "just-the-docs", "~> 0.3"
+gem "just-the-docs", "0.6.2"
```
Comment (nonrational) [inline Gemfile:4]:
```suggestion
# when upgrading, ensure this version matches `remote_theme` in _config.yml
gem "just-the-docs", "0.6.2"
```

Is it necessary to keep these in sync? I would not automatically know to go looking in `_config.yml` when attempting to upgrade this.

**Review by nonrational — APPROVED** [review]
🔍 🙌

**Review by bmuller — APPROVED** [review]

**Comment by bmuller** [comment]
@pavelloz I'll go ahead and merge 🤞

**Comment by pavelloz** [comment]
Looks like it fixed the issue, 👍

**Pull request merged**